### PR TITLE
[demo] support custom default pod label in pod

### PR DESCRIPTION
**Pull request by Frapschen**
support costum default pod label for istio sidecar injection case

close https://github.com/open-telemetry/opentelemetry-helm-charts/issues/750

### CODE REVIEW

**Review by TylerHelmuth — APPROVED** [review]

File: charts/opentelemetry-demo/values.yaml [hunk 1]
```diff
@@ -48,6 +48,9 @@ default:
     affinity: {}
     tolerations: []
   securityContext: {}
+  # list of pod labels applied to all components
+  podLabels: {}
+  # sidecar.istio.io/inject: true
```
Comment (dmitryax) [inline charts/opentelemetry-demo/values.yaml:53]: If you need it for istio, I believe the best practice is using annotations, not labels. I didn't even know that labels supported the same way

**Comment by puckpuck** [comment]
Istio support was https://github.com/open-telemetry/opentelemetry-helm-charts/pull/715 about a month ago. Can you do a helm repo update and try again?

File: charts/opentelemetry-demo/values.yaml (hunk 1 above)
Comment (Frapschen) [reply charts/opentelemetry-demo/values.yaml:53]: I hope I can use annotations, but I found this way is [Deprecated](https://istio.io/latest/docs/reference/config/annotations/). [Controlling the injection policy](https://istio.io/latest/docs/setup/additional-setup/sidecar-injection/#controlling-the-injection-policy) tell me to use lables.

**Review by TylerHelmuth — APPROVED** [review]

**Comment by TylerHelmuth** [comment]
@puckpuck please take another look

**Pull request closed without merging**